python/lsst/obs/fiberspectrograph/translator.py

```diff
@@ -0,0 +1,162 @@
+
+    default_resource_root = os.path.join(getPackageDir("obs_fiberspectrograph"), "corrections")
+    """Default resource path root to use to locate header correction files."""
+
+    DETECTOR_MAX = 1
+
+    _const_map = {
+        "detector_num": 0,
+        "detector_name": "0",
+        "object": None,
+        "physical_filter": "empty",
+        "detector_serial": "1606191U1",
+        "detector_group": "None",
+        "relative_humidity": None,
+        "pressure": None,
+        "temperature": None,
+        "focus_z": None,
+        "boresight_airmass": None,
+        "boresight_rotation_angle": None,
+        "tracking_radec": None,
+        "telescope": SIMONYI_TELESCOPE,
+        "observation_type": "spectrum",  # IMGTYPE is ''
+    }
+    """Constant mappings"""
+
+    _trivial_map = {
+        "observation_id": "OBSID",
+        "science_program": ("PROGRAM", dict(default="unknown")),
+    }
+    """One-to-one mappings"""
+
+    @classmethod
+    def can_translate(cls, header, filename=None):
+        """Indicate whether this translation class can translate the
+        supplied header.
+
+        Parameters
+        ----------
+        header : `dict`-like
+            Header to convert to standardized form.
+        filename : `str`, optional
+            Name of file being translated.
+
+        Returns
+        -------
+        can : `bool`
+            `True` if the header is recognized by this class. `False`
+            otherwise.
+        """
+
+        return "INSTRUME" in header and header["INSTRUME"] in ["FiberSpectrograph.Broad"]
+
+    @cache_translation
+    def to_instrument(self):
+        return "FiberSpec"
+
+    @cache_translation
+    def to_datetime_begin(self):
+        self._used_these_cards("DATE-BEG")
+        return Time(self._header["DATE-BEG"], scale="tai", format="isot")
+
+    @cache_translation
+    def to_exposure_time(self):
+        # Docstring will be inherited. Property defined in properties.py
+        # Some data is missing a value for EXPTIME.
+        # Have to be careful we do not have circular logic when trying to
+        # guess
+        if self.is_key_ok("EXPTIME"):
+            return self.quantity_from_card("EXPTIME", u.s)
+
+        # A missing or undefined EXPTIME is problematic. Set to -1
+        # to indicate that none was found.
+        log.warning("%s: Insufficient information to derive exposure time. Setting to -1.0s",
+                    self._log_prefix)
+        return -1.0 * u.s
+
+    @cache_translation
+    def to_dark_time(self):             # N.b. defining this suppresses a warning re setting from exptime
+        if "DARKTIME" in self._header:
+            darkTime = self._header["DARKTIME"]
+            self._used_these_cards("DARKTIME")
+            return (darkTime, dict(unit=u.s))
+        return self.to_exposure_time()
+
+    @staticmethod
+    def compute_exposure_id(dayobs, seqnum, controller=None):
```

Comment:
How does this differ to the parent class implementation?

Reply:
The parent method requires a supported controller, but currently the controller in the header is not supported. We are postponing this update and all related header updates to DM-43041.